staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go

```diff
@@ -118,7 +122,17 @@ func (p *Patcher) patchSimple(obj runtime.Object, modified []byte, namespace, na
 	var patchType types.PatchType
 	var patch []byte
 
-	if p.OpenapiSchema != nil {
+	if p.OpenAPIV3Root != nil {
+		// TODO: Capture error and log maybe?
+		if gvkSupported, _ := p.gvkSupportsPatchOpenAPIV3(p.Mapping.GroupVersionKind); gvkSupported {
+			patch, err = p.buildStrategicMergePatchFromOpenAPIV3(original, modified, current)
+			if err != nil {
+				fmt.Fprintf(errOut, "warning: error calculating patch from openapi v3 spec: %v\n", err)
+			}
+		}
+	}
+
+	if patch == nil && p.OpenapiSchema != nil {
```

Comment:
One problem I have with this code is that `p.OpenapiSchema` has already downloaded the OpenAPI v2. I think there's a different interface somewhere that allows a lazy fetching of OpenAPI v2. Otherwise we will always download both in apply.

Reply:
Not yet, we'll be reviving https://github.com/kubernetes/kubernetes/pull/116565 to lazy fetch OpenAPI V2, but I think we can proceed on this in parallel? And yes, this will currently download both v2 and v3 which the linked PR should fix.